### ISSUE TITLE
Prevent duplicates

### ISSUE THREAD

**Issue by iver56** [lionleaf/dwitter#53]
If a user tries to submit a dweet that already exists (i.e. the code is the same), then that dweet should not be accepted as a new dweet, and the user should get a message explaining why

**Comment by lionleaf** [comment]
As protection from blatant dweet-stealing or some other reason? Is this a problem at the moment?

**Comment by iver56** [comment]
It's not a big issue, but yes, it has happened a few times

**Comment by lionleaf** [comment]
But are those cases bad? Are you referring to your own repost, or someone else literally copy pasting another dweet without change? And does it matter if there are equal dweets out there?

**Comment by iver56** [comment]
I created this issue when somebody reposted one of my dweets. It was identical and just above my dweet in the "new" stream, hence I didn't feel that the duplicate dweet added any value and I deleted it (as a member of the moderator staff). My thinking went something like "we should automate this" and I created this issue, thinking it would be easy to implement.

The feature also prevents

* erroneous double posts, which has not been an issue yet (AFAIK)
* people from posting the example code (which has happened)

**Comment by lionleaf** [comment]
I agree that this is unfortunate. However, it will be easy to get past a block on equal dweets.

A better solution might be a "This dweet was first posted by: iverjo  *link to dweet*"  or something similar.

**Comment by iver56** [comment]
This is still a problem. I have recently deleted 4-5 duplicates

**Comment by lionleaf** [comment]
Duplicates of their own dweets or someone elses? What is the cause of this do you think? Is it intentional? Same person? New people experimenting with the site? People wanting to steal credit could easily get past a filter by changing small things

**Comment by iver56** [comment]
Mainly duplicates of the default template and duplicates of other users' dweets. Several users have done this. I think it's just people experimenting with the site. Don't think they are trying to steal credit.
